packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -254,16 +259,27 @@ - (void)updateOrientation:(UIDeviceOrientation)orientation
 
 - (void)captureToFile:(FLTThreadSafeFlutterResult *)result {
   AVCapturePhotoSettings *settings = [AVCapturePhotoSettings photoSettings];
+  NSString *extension = @"jpg";
   if (_resolutionPreset == FLTResolutionPresetMax) {
     [settings setHighResolutionPhotoEnabled:YES];
   }
 
+  if (_outputFormat == OutputFormatHEIF) {
+    if (@available(iOS 11.0, *)) {
+      if ([self.capturePhotoOutput.availablePhotoCodecTypes containsObject:AVVideoCodecTypeHEVC]) {
+        settings = [AVCapturePhotoSettings
+            photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeHEVC}];
+        extension = @"heif";
```

Comment:
rather than setting it to `jpg` and then overwrite it to `heif`, can you write `if/else`?

Reply:
what prevents us from writing

```
if supported {
  extension = "heif"
else {
  extension = "jpg"
}
```

Reply:
I rewrote it to use if/else